src/nodeodm_client/nodeodm_client.py

```diff
@@ -0,0 +1,237 @@
+import asyncio
+import json
+import logging
+import logging.config
+import os
+
+import requests
+
+
+def switch_case(status_code):
+    switcher = {
+        10: "Task queued.",
+        20: "Task running.",
+        30: "Task failed.",
+        40: "Task completed.",
+        50: "Task canceled."
+    }
+    return switcher.get(status_code, "Invalid status code")
+
+
+class NodeodmClient:
+    """ A class to interact with the NodeODM API.
+
+    """
+
+    def __init__(self, image_dir, nodeodm_url):
+        logging.config.fileConfig('logging.conf')
+        self.logger = logging.getLogger('app')
+        temp = os.path.dirname(os.path.abspath(__file__))
+        self.project_dir = os.path.dirname(temp)
+        self.image_dir = image_dir
+        self.nodeodm_url = nodeodm_url
+        self.CHUNK_SIZE = 20
+
+    def http_get(self, url):
```

Comment:
Die Kapselung hat den Nachteil, dass du fast in allen Methoden noch mal eigenes Fehlerhandling machst. D.h. so richtig viel gewinnst du hier nicht. Des Weiteren hast du den Nachteil, dass du so keine individuellen Header mitgeben kannst, ich würde die Methodenaufrufe daher nicht kapseln.